app/gui/src/dashboard/providers/SessionProvider/SessionProvider.tsx

```diff
@@ -289,6 +289,14 @@ export function SessionProvider(props: SessionProviderProps) {
     }
   }, [session.data, saveAccessTokenEventCallback])
 
+  React.useEffect(() => {
```

Comment:
Just FYI, there's a article on how to refresh sessions based on 401: https://dev.to/elmehdiamlou/efficient-refresh-token-implementation-with-react-query-and-axios-f8d